src/data/nat/factorization.lean

```diff
@@ -125,4 +125,41 @@ begin
   rw mem_factors_mul_of_pos ha.bot_lt hb.bot_lt,
 end
 
+/-- For any multiplicative function `f` with `f 1 = 1` and any `n > 0`,
+we can evaluate `f n` by evaluating `f` at `p ^ k` over the factorization of `n` -/
+lemma multiplicative_factorization {β : Type*} [comm_monoid β] (f : ℕ → β)
+  (h_mult : ∀ x y : ℕ, coprime x y → f (x * y) = f x * f y) (hf : f 1 = 1) :
+  ∀ {n : ℕ}, 0 < n → f n = n.factorization.prod (λ p k, f (p ^ k)) :=
```

Comment:
Probably n ne zero would be better here, and below, for the same reasons as usual (https://leanprover-community.github.io/contribute/style.html#normal-forms)

Reply:
Ah, thanks.  I keep being confused about which convention to use and should have thought to consult the style guide.

Reply:
This one is tricky because it's dependent on whether it's an "output" or an "input" of the lemma in question! No worries :)